### PR TITLE
support parse mysql administration statements sql SHOW TRIGGERS.(#13143)

### LINKED ISSUE #13143: support parsing SHOW TRIGGERS for mysql
Hi community,

This issue can be a good first try.

## Aim
support parse mysql administration statements sql `SHOW TRIGGERS`

## Basic Qualifications
- java
- maven
- antlr4

## Detail
We have completed the clone rule in DALStatement.g4.  refer to https://dev.mysql.com/doc/refman/8.0/en/show-triggers.html
But have not yet implement the visit method in `shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/impl/MySQLDALStatementSQLVisitor.java`
To complete this issue, I think we should complete these steps:
- Create `MySQLShowTriggersStatement` class in this package `org.apache.shardingsphere.sql.parser.sql.dialect.statement.mysql.dal`
- Add rule in `shardingsphere-sql-parser/shardingsphere-sql-parser-engine/src/main/java/org/apache/shardingsphere/sql/parser/core/database/visitor/SQLVisitorRule.java`
- Implement the visit method in `shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/mysql/visitor/statement/impl/MySQLDALStatementSQLVisitor.java`. 
- Finally, add test for parse clone statement in `shardingsphere-test` module
## Example FYI
- Refer to this merged pr：https://github.com/apache/shardingsphere/pull/12979/files

### PR DESCRIPTION
**Pull request by clbigdata**
Fixes #13143.

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/statement/dal/ShowTriggersStatementTestCase.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.shardingsphere.test.sql.parser.parameterized.jaxb.cases.domain.segment.impl.like.ExpectedLikeClause;
+import org.apache.shardingsphere.test.sql.parser.parameterized.jaxb.cases.domain.segment.impl.schema.ExpectedSchema;
+import org.apache.shardingsphere.test.sql.parser.parameterized.jaxb.cases.domain.statement.SQLParserTestCase;
+
+import javax.xml.bind.annotation.XmlElement;
+
+/**
+ * Show tables statement test case.
```
Comment (tuichenchuxin) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/statement/dal/ShowTriggersStatementTestCase.java:29]: May be show triggers better

**Review by tuichenchuxin — COMMENTED** [review]
Welcome. Please resovle the conflicat.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/dal/MySQLShowTriggersStatement.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    /**
+     * Get like segment.
+     *
+     * @return like segment
+     */
+    public Optional<ShowLikeSegment> getLike() {
+        return Optional.ofNullable(like);
+    }
+
```
Comment (dongzl) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/dal/MySQLShowTriggersStatement.java:59]: remove blank line.

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/dal/impl/ShowTriggersStatementAssert.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    private static void assertLike(final SQLCaseAssertContext assertContext, final MySQLShowTriggersStatement actual, final ShowTriggersStatementTestCase expected) {
+        if (null != expected.getLike()) {
+            assertTrue(assertContext.getText("Actual like should exist."), actual.getLike().isPresent());
+            SQLSegmentAssert.assertIs(assertContext, actual.getLike().get(), expected.getLike());
+        } else {
+            assertFalse(assertContext.getText("Actual like should not exist."), actual.getLike().isPresent());
+        }
+    }
+
```
Comment (dongzl) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/dal/impl/ShowTriggersStatementAssert.java:65]: remove blank line.

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/dal/show.xml [hunk 4]
```diff
@@ -102,4 +102,10 @@
     <show-replica-status sql-case-id="show_replica_status_with_channel" channel="TEST_CHANNEL"/>
     <show-slave-status sql-case-id="show_slave_status" />
     <show-slave-status sql-case-id="show_slave_status_with_channel" channel="TEST_CHANNEL" />
+
```
Comment (dongzl) [inline shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/dal/show.xml:105]: remove blank line.

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/dal/show.xml [hunk 5]
```diff
@@ -102,4 +102,10 @@
...
+    <show-triggers sql-case-id="show_triggers_with_filter">
+        <schema name="sharding_db" start-delimiter="`" end-delimiter="`" start-index="14" stop-index="31" />
+        <like pattern="acc%" start-index="33" stop-index="43"/>
+    </show-triggers>
+
```
Comment (dongzl) [inline shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/dal/show.xml:110]: the same to above.

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/statement/dal/ShowTriggersStatementTestCase.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+@Getter
+@Setter
+public final class ShowTriggersStatementTestCase extends SQLParserTestCase {
+
+    @XmlElement
+    private ExpectedSchema schema;
+
+    @XmlElement
+    private ExpectedLikeClause like;
+
```
Comment (dongzl) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/statement/dal/ShowTriggersStatementTestCase.java:40]: the same.

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/SQLParserTestCases.java [hunk 7]
```diff
@@ -754,7 +758,7 @@
 
     @XmlElement(name = "shutdown")
     private final List<ShutdownStatementTestCase> shutdownStatementTestCases = new LinkedList<>();
-
```
Comment (tuichenchuxin) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/SQLParserTestCases.java:757]: Please keep the code style same.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/dal/MySQLShowTriggersStatement.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+ */
+@Setter
+@ToString
+public final class MySQLShowTriggersStatement extends AbstractSQLStatement implements DALStatement, MySQLStatement {
+
+    private FromSchemaSegment fromSchema;
+
+    private ShowLikeSegment like;
+
+
```
Comment (tuichenchuxin) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/dal/MySQLShowTriggersStatement.java:41]: remove useless empty line.

**Review by tuichenchuxin — COMMENTED** [review]
I think it's almost ok. Just modify the code style is enough.

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/dal/impl/ShowTriggersStatementAssert.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+import org.apache.shardingsphere.test.sql.parser.parameterized.jaxb.cases.domain.statement.dal.ShowTriggersStatementTestCase;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Show triggers statement assert.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class ShowTriggersStatementAssert {
```
Comment (strongduanmu) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/asserts/statement/dal/impl/ShowTriggersStatementAssert.java:34]: @clbigdata Please add final for ShowTriggersStatementAssert.

File: shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/SQLParserTestCases.java [hunk 10]
```diff
@@ -703,7 +707,7 @@
     
     @XmlElement(name = "show-single-table")
     private final List<ShowSingleTableStatementTestCase> showSingleTableStatementTestCases = new LinkedList<>();
-
```
Comment (strongduanmu) [inline shardingsphere-test/shardingsphere-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/jaxb/cases/domain/SQLParserTestCases.java:706]: @clbigdata Please keep indents consistent with the previous one.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Review by strongduanmu — APPROVED** [review]
@clbigdata Good job!!! 👏

**Pull request merged**